lib/python/picongpu/plugins/data/emittance.py

```diff
@@ -0,0 +1,181 @@
+            self.data_file_suffix
+        )
+        if not os.path.isfile(data_file_path):
+            raise IOError('The file {} does not exist.\n'
+                          'Did the simulation already run?'
+                          .format(data_file_path))
+
+        return data_file_path
+
+    def get_iterations(self, species, species_filter="all"):
+        """
+        Return an array of iterations with available data.
+
+        Parameters
+        ----------
+        species : string
+            short name of the particle species, e.g. 'e' for electrons
+            (defined in ``speciesDefinition.param``)
+        species_filter: string
+            name of the particle species filter, default is 'all'
+            (defined in ``particleFilters.param``)
+
+        Returns
+        -------
+        An array with unsigned integers.
+        """
+        data_file_path = self.get_data_path(species, species_filter)
+
+        # the first column contains the iterations
+        return pd.read_csv(data_file_path,
+                           usecols=(0,),
+                           delimiter=" ",
+                           dtype=np.uint64).values[:, 0]
+
+    def _get_for_iteration(self, iteration, species,
+                           species_filter="all", **kwargs):
+        """
+        Get a histogram.
+
+        Parameters
+        ----------
+        species : string
+            short name of the particle species, e.g. 'e' for electrons
+            (defined in ``speciesDefinition.param``)
+        species_filter: string
+            name of the particle species filter, default is 'all'
+            (defined in ``particleFilters.param``)
+        iteration : (unsigned) int [unitless]
+            The iteration at which to read the data.
+            A list of iterations is allowed as well.
+            ``None`` refers to the list of all available iterations.
+        sum : float
+            emittance value [m rad] without slicing
+
+        Returns
+        -------
+        slice_emit : np.array of float
+            slice emittance [m rad] for each y_slice
+            If iteration is a list, returns (ordered) dict with
+            iterations as its index.
+        y_slices : np.array of float
+            beginning of each slice [m]
+        iteration : (unsigned) int [unitless]
+            The iteration at which to read the data.
+            A list of iterations is allowed as well.
+        """
+        if iteration is not None:
+            if not isinstance(iteration, collections.Iterable):
+                iteration = np.array([iteration])
+
+        data_file_path = self.get_data_path(species, species_filter)
+
+        # read whole file as pandas.DataFrame
+        data = pd.read_csv(
+            data_file_path,
+            delimiter=" "
+        )
+
+        # note: only reads first row and selects the valid emittance slices
+        y_slices = pd.read_csv(
```

Comment:
Is it necessary to read the file again? You already have the full file content within the `data` variable.

Reply:
it is the same as in energy_histogram.py

Reply:
this only reads one line which is ginored in the read above since it's a comment :)